src/html/util.ts

```diff
@@ -173,27 +173,34 @@ export function getIndentationInside(parentNode: dom5.Node) {
   return match[2];
 }
 
+export interface AttributeAliases { [s: string]: string; }
+
 /**
  * Converts a css selector into a dom5 predicate.
  *
  * This is intended for handling only selectors that match an individual element
  * in isolation, it does throws if the selector talks about relationships
  * between elements like `.foo .bar` or `.foo > .bar`.
+ *
+ * Pass `attributeAliases` to apply selectors in non-standard attributes, e.g.
+ * <div class$="foo bar"></div>
  */
-export function elementSelectorToPredicate(simpleSelector: string):
-    dom5.Predicate {
+export function elementSelectorToPredicate(
+    simpleSelector: string,
+    attributeAliases: AttributeAliases = {}): dom5.Predicate {
```

Comment:
Hm, there are other cases where we'd like this behavior, and the attributeAliases syntax requires more knowledge from the caller.

What we want to say here is that if we're inside a Polymer template databinding context, then we want attribute matching to be slightly different. So if we had a boolean that indicated that we're in a polymer template, then for some of the attribute selectors (maybe `exists`, `element`, and `any`?) we would match on both the attribute name and the attribute name with a `$` suffix

Reply:
Updated to append $ for polymer templates. I did it for all actions, as we might want to have selectors like `[src^="//www.lorempixel.com/"]`, which corresponds to the action `start`, and should return nodes that are bound like this:
```html
<img src$="//www.lorempixel.com/[[dimension]]/[[dimension]]">
```